engines/bops_core/app/controllers/concerns/bops_core/magic_link_authenticatable.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module BopsCore
+  module MagicLinkAuthenticatable
+    extend ActiveSupport::Concern
+
+    SGID_FOR = "magic_link"
+
+    included do
+      rescue_from ActionController::ParameterMissing do |exception|
+        render plain: "Not Found", status: :not_found
+      end
+    end
+
+    def authenticate_with_sgid!
+      resource = GlobalID::Locator.locate_signed(sgid, for: SGID_FOR)
```

Comment:
I don't think we really need the constant for this.

```suggestion
    included do
      rescue_from ActionController::ParameterMissing do |exception|
        render plain: "Not Found", status: :not_found
      end
    end

    def authenticate_with_sgid!
      resource = GlobalID::Locator.locate_signed(sgid, for: "magic_link")
```

Reply:
Yeah, I saw there were a few other places in other files that used the literal string that could in theory be changed to use the constant but in practice I think it's probably not worth even that.

Reply:
Now that I've added a check if the sgid has expired with base64 decoding, need to check against the specified sgid for so leaving this as a constant